### PR TITLE
Fix#1930 Added fade filter with BW contrast

### PR DESCRIPTION
**Pull request by Pravesh-Jamgade**
Fixed https://github.com/fossasia/phimpme-android/issues/1930

**Changes**:
I added fade filter.It creates greyscale image and perform bw contrast.
I changed files from **phimpme-android/app/src/main/jni/** location containing filters.cpp   ,filters.h , main_processing.cpp and an array file from **app/src/main/res/values/arrays.xml** .

**Screenshots for the change**:
**Before** 
image
**After** 
image

### CODE REVIEW

File: app/src/main/jni/filters.cpp [hunk 1]
```diff
@@ -534,5 +534,25 @@ void applyCyano(cv::Mat &src, cv::Mat &dst, int val) {
...
+    {
+        for(int x=0 ;x < src.rows ; x++) {
+            Vec3b intensity = src.at<Vec3b>(y, x);
+            uchar blue = intensity.val[0];
+            uchar green = intensity.val[1];
+            uchar red = intensity.val[2];
+
+            dst.at<Vec3b>(y, x)[0] = saturate_cast<uchar> (blue + 0.5 *val );
+            dst.at<Vec3b>(y, x)[1] = saturate_cast<uchar> (green + 0.5 *val);
+            dst.at<Vec3b>(y, x)[2] = saturate_cast<uchar> (red + 0.5 *val) ;
```
Comment (mayank8318) [inline app/src/main/jni/filters.cpp]: Please follow proper indentation. Avoid extra spaces and don't put spaces before a semi-colon. Check this in the loop condition as well.

File: app/src/main/jni/filters.h [hunk 2]
```diff
@@ -119,10 +119,12 @@ void applyThreshold(cv::Mat &inp, cv::Mat &out, int val);
 void applyGrain(cv::Mat &inp, cv::Mat &out, int val);
 void applyHistEq(cv::Mat &inp, cv::Mat &out, int val);
 void applyCyano(cv::Mat &inp, cv::Mat &out, int val);
+void fade(cv::Mat &src, cv::Mat &dst, int val);
```
Comment (mayank8318) [inline app/src/main/jni/filters.h]: Please follow the convention. This should be `applyFade`

File: app/src/main/res/values/arrays.xml [hunk 3]
```diff
@@ -684,8 +684,8 @@
         <item>Blue Boost</item>
         <item>Colour Enhance</item>
         <item>Cyanisation</item>
-
-  </string-array>
+        <item>fade</item>
```
Comment (mayank8318) [inline app/src/main/res/values/arrays.xml]: `Fade`

Comment (Pravesh-Jamgade) [reply app/src/main/res/values/arrays.xml]: thank you @mayank8318

Comment (Pravesh-Jamgade) [reply app/src/main/res/values/arrays.xml]: a little help needed @mayank8318 ,from where should i made these changes.Do i need to close these PR and make new one again

**Comment by Pravesh-Jamgade** [comment]
@mayank8318 need help here

**Comment by Pravesh-Jamgade** [comment]
@mayank8318 i did squash the commits by rebasing (git rebase -i origin/development)
it open the file
`
# Rebase 8b0dbdab..8b0dbdab onto 8b0dbdab (1 command)
#
# Commands:
# p, pick = use commit
# r, reword = use commit, but edit the commit message
# e, edit = use commit, but stop for amending
# s, squash = use commit, but meld into previous commit
# f, fixup = like "squash", but discard this commit's log message
# x, exec = run command (the rest of the line) using shell
# d, drop = remove commit
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
`
but file  should show the message like these
`pick 01d1124 Message....
pick 6340aaa Message....
pick ebfd367 Message....
pick 30e0ccb Message....
`
is there something wrong??

**Comment by Pravesh-Jamgade** [comment]
@mayank8318 squashing should be done on my working branch or on the master of project???

**Comment by mayank8318** [comment]
Squashing is done on your working branch. And you need to use the `squash` command, not `rebase`

**Comment by Pravesh-Jamgade** [comment]
@mayank8318 please review again

**Pull request closed without merging**